cmd/encryption-v1.go

```diff
@@ -623,22 +623,22 @@ func DecryptBlocksRequest(client io.Writer, r *http.Request, startOffset, length
 	var partStartOffset = startOffset
 	// Skip parts until final offset maps to a particular part offset.
 	for i, part := range objInfo.Parts {
-		decryptedSize, err := decryptedSize(part.Size)
+		decryptedSize, err := sio.DecryptedSize(uint64(part.Size))
 		if err != nil {
-			return nil, -1, -1, err
+			return nil, -1, -1, errObjectTampered
```

Comment:
Not sure if its okay to assume `errObjectTampered` for all `err` thrown by `DecryptedSize`?

Reply:
It is fine - `DecryptedSize` can fail for two reasons:
 1. The size is greater than `2^48 + 2^37` which cannot happen if the object is a valid S3 object. `2^48 + 2^37 > 256 TB` - S3 allows only 5 TB
 2. The size would imply a last package which is smaller than 33 bytes. That cannot happen if the object is a valid encrypt object.

-> The object is accidentally/maliciously tampered.